app/assets/javascripts/admin/controllers/admin-dashboard-general.js

```diff
@@ -23,9 +23,44 @@ export default Controller.extend(PeriodComputationMixin, {
 
   @discourseComputed("siteSettings.dashboard_general_tab_activity_metrics")
   activityMetrics(metrics) {
-    return (metrics || "").split("|").filter(m => m);
+    return (metrics || "").split("|").filter(Boolean);
   },
 
+  hiddenReports: computed("siteSettings.dashboard_hidden_reports", function() {
+    return (this.siteSettings.dashboard_hidden_reports || "")
+      .split("|")
+      .filter(Boolean);
+  }),
+
+  isActivityMetricsVisible: computed(
+    "activityMetrics",
+    "hiddenReports",
+    function() {
+      return (
+        this.activityMetrics.length > 0 &&
+        !this.hiddenReports.any(x => this.activityMetrics.includes(x))
+      );
+    }
+  ),
+
+  isSearchReportsVisible: computed("hiddenReports", function() {
+    return ["top_referred_topics", "trending_search"].filter(
+      x => !this.hiddenReports.includes(x)
+    ).length;
+  }),
+
+  isCommunityHealthVisible: computed("hiddenReports", function() {
+    return [
+      "consolidated_page_views",
+      "signups",
+      "topics",
+      "posts",
+      "dau_by_mau",
+      "daily_engaged_users",
+      "new_contributors"
+    ].filter(x => !this.hiddenReports.includes(x)).length;
```

Comment:
I would recommend using `.some` for this. No point looping through them all if you only care if one is visible.